bentoml/_internal/service/utils.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+
+def _cpu_converter(cpu: Union[int, float, str]) -> float:
+    if isinstance(cpu, (int, float)):
+        return float(cpu)
+
+    if isinstance(cpu, str):
+        milli_match = re.match("([0-9]+)m", cpu)
+        if milli_match:
+            return int(milli_match[1]) / 1000.0
+
+    raise ValueError(f"Invalid CPU resource limit '{cpu}'")
+
+
+def _mem_converter(mem: Union[int, str]) -> int:
+    if isinstance(mem, int):
+        return mem
+
+    if isinstance(mem, str):
+        unit_match = re.match("([0-9]+)([A-Za-z]{1,2})", mem)
+        mem_multipliers = {
+            k: 1000,
+            M: 1000 ** 2,
+            G: 1000 ** 3,
+            T: 1000 ** 4,
+            P: 1000 ** 5,
+            E: 1000 ** 6,
+            Ki: 1024,
+            Mi: 1024 ** 2,
+            Gi: 1024 ** 3,
+            Ti: 1024 ** 4,
+            Pi: 1024 ** 5,
+            Ei: 1024 ** 6,
+        }
+        if unit_match:
+            base = int(unit_match[1])
+            unit = unit_match[2]
+            if unit in mem_multipliers:
+                return base * mem_multipliers[unit]
+
+    raise ValueError(f"Invalid MEM resource limit '{mem}'")
+
+
+def _query_cpu_count() -> float:
```

Comment:
wrt memory management, we should also consider to use shared memory on Linux system via `/dev/shm` for better performance.